### PR TITLE
docs: reorganize (part 2)

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
The current documentation is difficult to understand where it is written.
See #839

- add section "User Management"
- add pages
- move files to the section directories
- tweaks

Screenshot 2023-09-19 10 53 02

Online Demo: https://datamweb.github.io/shield/

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: docs/quick_start_guide/using_session_auth.md [hunk 1]
```diff
@@ -1,11 +1,13 @@
-# Authentication Flow
+# Using Session Authenticator
+
+**Session** authenticator provides traditional Email/Password authentication.
```
Comment (datamweb) [inline docs/quick_start_guide/using_session_auth.md:3]: **Email/Password** It's probably a term, but my understanding of this at first glance is that Shield only supports email/password (username/password, mobile/password, etc.).

File: docs/references/magic_link_login.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Magic Link Login
+
+Magic Link Login is a feature that allows users to log in if they forget their
+password.
```
Comment (datamweb) [inline docs/references/magic_link_login.md:4]: Or use as a one-time password.

**Review by datamweb — APPROVED** [review]
Generally good.
Only I prefer to have section [acknowledgements](https://github.com/codeigniter4/shield#acknowledgements) in these documents.

File: docs/getting_started/install.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    #### Note: migration error
+
+    When you run `spark migrate --all`, if you get `Class "SQLite3" not found` error:
+
+    1. Remove sample migration files in **tests/_support/Database/Migrations/**
+    2. Or install `sqlite3` php extension
+
+    If you get `Specified key was too long` error:
+
+    1. Use InnoDB, not MyISAM.
```
[the remark this answers is not in the document]
Comment (kenjis) [reply docs/getting_started/install.md]: Removed.

File: docs/getting_started/install.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+6. Configure **app/Config/Email.php** to allow Shield to send emails.
+
+    ```php
+    <?php
```
[the remark this answers is not in the document]
Comment (kenjis) [reply docs/getting_started/install.md:162]: This is a whole file of it. So it seems better to have `<?php`.
I don't think `<?php` should be added to all existing codes.

File: docs/quick_start_guide/using_session_auth.md (hunk 1 above)
Comment (kenjis) [reply docs/quick_start_guide/using_session_auth.md]: Changed to `ID/Password`.

File: docs/quick_start_guide/using_session_auth.md [hunk 5]
```diff
@@ -27,7 +29,7 @@ public array $redirects = [
 
 ### Configure Remember-me Functionality
 
-Remember-me functionality is enabled by default for the `Session` authenticator. While this is handled in a secure manner, some sites may want it disabled. You might also want to change how long it remembers a user and doesn't require additional login.
+Remember-me functionality is enabled by default. While this is handled in a secure manner, some sites may want it disabled. You might also want to change how long it remembers a user and doesn't require additional login.
```
[the remark this answers is not in the document]
Comment (kenjis) [reply docs/quick_start_guide/using_session_auth.md:32]: Remember-me. Done.

File: docs/references/controller_filters.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+| group       | Checks if the user is in one of the groups passed in.                                                                                                                                                                               |
+| permission  | Checks if the user has the passed permissions.                                                                                                                                                                                      |
+| force-reset | Checks if the user requires a password reset.                                                                                                                                                                                       |
+
+These can be used in any of the [normal filter config settings](https://codeigniter.com/user_guide/incoming/filters.html#globals), or [within the routes file](https://codeigniter.com/user_guide/incoming/routing.html#applying-filters).
+
+> **Note**
+> These filters are already loaded for you by the registrar class located at **src/Config/Registrar.php**.
```
[the remark this answers is not in the document]
Comment (kenjis) [reply docs/references/controller_filters.md]: Moved the note up.

File: docs/references/magic_link_login.md (hunk 2 above)
Comment (kenjis) [reply docs/references/magic_link_login.md:4]: Using Magic Link Login as a one-time password authentication?
I don't know if it is supposed to be used for anything other than forgotten passwords.

**Comment by kenjis** [comment]
Added Acknowledgements.

**Pull request merged**